tests/src/test/scala/ironrust/FullIntegrationTest.scala

```diff
@@ -205,8 +296,11 @@ class FullIntegrationTest extends DudeSuite with CancelAfterFailure {
         Try(IronOxide.generateNewDevice(jwt, primaryTestUserPassword, new DeviceCreateOpts, null)).toEither
       val dev3 = new DeviceContext(deviceResp2.value)
 
-      val deviceList = Try(sdk.userListDevices).toEither.value.getResult
+      val deviceListResult = Try(sdk.userListDevices).toEither.value
+      val deviceList = deviceListResult.getResult
 
+      deviceListResult.equals(deviceListResult) shouldBe true
```

Comment:
This kind of test could be written by doing 2 list calls back to back (or in parallel) and then comparing them. We can write a nice little helper function that will do that for anything. Something like `def twice[A](a: => A): IO[(A,A)]` should make the tests only slightly slower, while allowing us to do the comparisons.

Reply:
I went a different direction than this, so give it a look